sdcm/nemesis.py

```diff
@@ -4863,46 +4845,48 @@ def disrupt_add_remove_dc(self) -> None:
             raise UnsupportedNemesis(
                 "add_remove_dc skipped for multi-dc scenario (https://github.com/scylladb/scylla-cluster-tests/issues/5369)")
         InfoEvent(message='Starting New DC Nemesis').publish()
+
         node = self.cluster.data_nodes[0]
+        stress_tests = []
         system_keyspaces = ["system_distributed", "system_traces"]
-        if not node.raft.is_consistent_topology_changes_enabled:  # auth-v2 is used when consistent topology is enabled
-            system_keyspaces.insert(0, "system_auth")
-        self._switch_to_network_replication_strategy(self.cluster.get_test_keyspaces() + system_keyspaces)
+        if not node.raft.is_consistent_topology_changes_enabled:
+            system_keyspaces.append("system-auth")
+
         datacenters = list(self.tester.db_cluster.get_nodetool_status().keys())
         self.tester.create_keyspace("keyspace_new_dc", replication_factor={
                                     datacenters[0]: min(3, len(self.cluster.data_nodes))})
-        node_added = False
+
         with ExitStack() as context_manager:
             def finalizer(exc_type, *_):
                 # in case of test end/killed, leave the cleanup alone
                 if exc_type is not KillNemesis:
                     with self.cluster.cql_connection_patient(node) as session:
                         session.execute('DROP KEYSPACE IF EXISTS keyspace_new_dc')
-                    if node_added:
-                        self.cluster.decommission(new_node)
+                        if self.cluster.get_node_status_dictionary(new_node.ip_address):
+                            [st.kill() for st in stress_tests]
+                            self.cluster.dc_remove([new_node], system_keyspaces)
```

Comment:
you changed also logic here - if it fails after node is decommissioned, there's no point of removing dc